presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionSymbolInliner.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.spi.relation.LambdaDefinitionExpression;
+import com.facebook.presto.spi.relation.RowExpression;
+import com.facebook.presto.spi.relation.VariableReferenceExpression;
+import com.facebook.presto.sql.relational.RowExpressionRewriter;
+import com.facebook.presto.sql.relational.RowExpressionTreeRewriter;
+import com.facebook.presto.sql.tree.SymbolReference;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static com.facebook.presto.sql.relational.OriginalExpressionUtils.castToExpression;
+import static com.facebook.presto.sql.relational.OriginalExpressionUtils.castToRowExpression;
+import static com.facebook.presto.sql.relational.OriginalExpressionUtils.isExpression;
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.stream.Collectors.toMap;
+
+public final class RowExpressionSymbolInliner
```

Comment:
Mark `ExpressionSymbolInliner` as deprecated and separate this class together with its test to a standalone commit.